lib/player.js

```diff
@@ -685,6 +703,68 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
     if (mediaElement) {
       this.attach(mediaElement, /* initializeMediaSource= */ true);
     }
+
+    if (canvas) {
+      this.canvas_ = canvas;
+    }
+  }
+
+  /**
+   * Create a shaka.lcevc.Dil object
+   * @private
+   */
+  createDIL_() {
+    if (this.lcevcDil_ == null) {
+      this.lcevcDil_ = new shaka.lcevc.Dil(
+          /** @type {HTMLVideoElement} */ (this.video_),
+          this.canvas_,
+          this.lcevcDilConfig_,
+      );
+      if (this.mediaSourceEngine_) {
+        this.mediaSourceEngine_.updateLcevcDil(this.lcevcDil_);
+      }
+    }
+  }
+
+  /**
+   * Close a shaka.lcevc.Dil object
+   * @private
+   */
+  closeDIL_() {
+    if (this.lcevcDil_ != null) {
```

Comment:
This checks to see if `this.lcevcDil_` is non-null. However, each time you call `closeDil_`, you also check if `this.lcevcDil_` is non-null before calling it.
You can probably remove all of those checks, since they are unnecessary, and update the JSDoc to say that it closes the DIL "if present" (or similar).

Reply:
Edited the JSDoc for closeDil and remove all checks in player.js before calling closeDil